python/ray/tune/stopper/stopper.py

```diff
@@ -15,34 +16,42 @@ class Stopper(abc.ABC):
 
         >>> import time
         >>> from ray import air, tune
+        >>> from ray.air import session
         >>> from ray.tune import Stopper
         >>>
         >>> class TimeStopper(Stopper):
         ...     def __init__(self):
         ...         self._start = time.time()
-        ...         self._deadline = 5
+        ...         self._deadline = 5  # Stop all trials after 5 seconds
         ...
         ...     def __call__(self, trial_id, result):
         ...         return False
         ...
         ...     def stop_all(self):
         ...         return time.time() - self._start > self._deadline
-        >>>
+        ...
+        >>> def train_fn(config):
+        ...     for i in range(100):
+        ...         time.sleep(1)
+        ...         session.report({"iter": i})
+        ...
         >>> tuner = tune.Tuner(
-        ...     tune.Trainable,
-        ...     tune_config=tune.TuneConfig(num_samples=200),
-        ...     run_config=air.RunConfig(stop=TimeStopper())
+        ...     train_fn,
+        ...     tune_config=tune.TuneConfig(num_samples=2),
+        ...     run_config=air.RunConfig(stop=TimeStopper()),
         ... )
-        >>> tuner.fit()
-        == Status ==...
+        >>> print("[ignore]"); result_grid = tuner.fit()  # doctest: +ELLIPSIS
+        [ignore]...
+        >>> all(result.metrics["time_total_s"] < 6 for result in result_grid)
+        True
 
     """
 
-    def __call__(self, trial_id, result):
+    def __call__(self, trial_id: str, result: Dict[str, Any]) -> bool:
         """Returns true if the trial should be terminated given the result."""
         raise NotImplementedError
 
-    def stop_all(self):
+    def stop_all(self) -> bool:
```

Comment:
Why do we include these typing changes? Does it improve the docs somehow?

Trying to understand the risk to semantic behavior change, since product won't do tests on this.

Reply:
Oh, I just added these in to clean it up a bit. These were already the assumed argument/return types - they just didn't have type hints.